[YUNIKORN-1914] change display for resource lists

wusamzong opened the pull request:
### What is this PR for?
Display a limited amount of resources to avoid overcrowding the page, and provide "Less..." and "More..." buttons for users to choose the amount of data to display.

In addition, modify the existing approach to use Angular's native *ngFor for rendering data, instead of utilizing the current innerHTML method.


### What type of PR is it?
* [x] - Improvement

### What is the Jira issue?
* Open an issue on Jira https://issues.apache.org/jira/browse/YUNIKORN-1914/

### How should this be tested?

### Screenshots (if appropriate)
![{21F2005A-8D8F-4BED-8A2A-BC9767E15049}](https://github.com/apache/yunikorn-web/assets/48400525/ca033f1f-5dcd-4075-a2fe-fb4e9504a56e)

![{55ADCCD0-989B-43E0-8CD7-60EF2A62D6B3}](https://github.com/apache/yunikorn-web/assets/48400525/8457f6ae-6365-46da-a300-7593b37ba4a6)

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

craigcondit commented:
One idea I've had for this is rather than a Less / More link for each entry, we instead provide short vs. detail global views. We could have a single toggle button on the page that changes the display to/from showing just CPU/Memory to showing all resource types. This would allow compact display as the default, but allow expanding the entire list at once for a more "resource-oriented" view. Thoughts?

stu01509 commented on src/app/components/apps-view/apps-view.component.ts:
```diff
@@ -16,7 +16,7 @@
  */
 
-import { Component, OnInit, ViewChild, ElementRef } from '@angular/core';
+import { Component, OnInit, ViewChild, ElementRef} from '@angular/core';
```
Hi
I am new to this project and I'd like to offer some suggestions for your consideration.

This import could add whitespace and follow the current style.

```js
import { Component, OnInit, ViewChild, ElementRef } from '@angular/core';
```

stu01509 commented on src/app/components/apps-view/apps-view.component.ts:
```diff
@@ -61,6 +61,8 @@ export class AppsViewComponent implements OnInit {
   leafQueueList: DropdownItem[] = [];
   leafQueueSelected = '';
+  
+  cellToggle = new Map<string,boolean>();
```
```js
cellToggle = new Map<string, boolean>();
```

stu01509 commented on src/app/components/apps-view/apps-view.component.ts:
```diff
@@ -301,4 +303,22 @@ export class AppsViewComponent implements OnInit {
       this.removeRowSelection();
     }
   }
+
+  formatResources(colValue:string):string[]{
+    let result:string[]=colValue.split("<br/>")
+    return result;
+  }
+
+  getToggle(cellId: string):boolean{
+    return this.cellToggle.get(cellId)!
+  }
+
+  toggle(cellId: string){
+    this.cellToggle.set(cellId, !this.getToggle(cellId))
+  }
+
+  getCellId(element: any, colId: any):string{
+    let cellId =`${element.applicationId}-${colId}`
+    return cellId
+  }
```
Seems like `result` and `cellId` aren't modified, so we can use the `const` instead of `let`;
And also seems like missing some semicolons.

```js
 formatResources(colValue: string): string[] {
    const result: string[] = colValue.split("<br/>");
    return result;
  }

  getToggle(cellId: string): boolean {
    return this.cellToggle.get(cellId)!
  }

  toggle(cellId: string): void {
    this.cellToggle.set(cellId, !this.getToggle(cellId));
  }

  getCellId(element: any, colId: any): string {
    const cellId =`${element.applicationId}-${colId}`;
    return cellId;
  }
```

stu01509 commented on src/app/components/nodes-view/nodes-view.component.ts:
```diff
@@ -190,4 +192,22 @@ export class NodesViewComponent implements OnInit {
     this.fetchNodeListForPartition(this.partitionSelected);
   }
+
+  formatResources(colValue:string):string[]{
+    let result:string[]=colValue.split("<br/>")
+    return result;
+  }
+
+  getToggle(cellId: string):boolean{
+    return this.cellToggle.get(cellId)!
+  }
+
+  toggle(cellId: string){
+    this.cellToggle.set(cellId, !this.getToggle(cellId))
+  }
+
+  getCellId(element: any, colId: any):string{
+    let cellId =`${element.applicationId}-${colId}`
+    return cellId
+  }
```
Same previous comment

wusamzong commented:
Thank you, @craigcondit and @stu01509, for providing valuable feedback on the review. I have incorporated your suggestions and made the necessary code changes accordingly.

I have implemented a single toggle to display details for all tables. While I've kept the previous display style, I've also changed "More..." to "..." to make the table more concise. If users choose to click on the "..." indicator, the display across all tables will be updated through a unified toggle action.
![{BA9BEA1F-1A96-4E24-89AF-DDA78F7BEEF4}](https://github.com/apache/yunikorn-web/assets/48400525/15a201a5-3708-4ad2-b718-3e38ce47ee33)
![{7120A6B5-4BDE-4A80-9A19-19531ABBAFE1}](https://github.com/apache/yunikorn-web/assets/48400525/24e55797-e351-4ed0-9b15-a4824a688507)

craigcondit commented:
> I have implemented a single toggle to display details for all tables. While I've kept the previous display style, I've also changed "More..." to "..." to make the table more concise. If users choose to click on the "..." indicator, the display across all tables will be updated through a unified toggle action. 

I think if we have a global button for the toggle, we don't need the "More" or "Less" links at all (these just get confusing). The button itself could simply be a "..." or "expando" type of image, and probably right-justified and aligned with the Partition link (the screenshot given looks a little sloppy). 

I also think having Memory and CPU as the default view, with  the expanded view showing all resources makes the most sense. Nearly every task schedules based on only these two resources, so they are universally useful. The others are not as much.
 
One other note: It would probably make sense to remove anything that resolves to "n/a" from the list as that is just noise.

wusamzong commented:
I aligned the global button with the Partition link and placed it on the right-justified.
Also I tried removing any item that resolves to "n/a" but ran into some issues that didn't work yet, I'll open a Jira to continue the process!

craigcondit reviewed (approved):
+1 LGTM. @wusamzong please file a follow-up JIRA for the handling of n/a resources.
